runtime/bundles/org.eclipse.core.runtime/src/org/eclipse/core/internal/runtime/InternalPlatform.java

```diff
@@ -654,78 +649,37 @@ public void stop(BundleContext bundleContext) {
 	}
 
 	private void openOSGiTrackers() {
+		instanceLocation = createOpenTracker(Location.INSTANCE_FILTER);
+		userLocation = createOpenTracker(Location.USER_FILTER);
+		configurationLocation = createOpenTracker(Location.CONFIGURATION_FILTER);
+		installLocation = createOpenTracker(Location.INSTALL_FILTER);
+		logTracker = createOpenTracker(FrameworkLog.class);
+		platformTracker = createOpenTracker(PlatformAdmin.class);
+		contentTracker = createOpenTracker(IContentTypeManager.class);
+		preferencesTracker = createOpenTracker(IPreferencesService.class);
+		groupProviderTracker = createOpenTracker("(objectClass=" + IBundleGroupProvider.class.getName() + ")"); //$NON-NLS-1$ //$NON-NLS-2$
+		logReaderTracker = createOpenTracker(ExtendedLogReaderService.class);
+		extendedLogTracker = createOpenTracker(ExtendedLogService.class);
+		environmentTracker = createOpenTracker(EnvironmentInfo.class);
+		debugTracker = createOpenTracker(DebugOptions.class);
+	}
+
+	private <T> ServiceTracker<T, T> createOpenTracker(String filterStr) {
 		Filter filter = null;
 		try {
-			filter = context.createFilter(Location.INSTANCE_FILTER);
-		} catch (InvalidSyntaxException e) {
-			// ignore this.  It should never happen as we have tested the above format.
-		}
-		instanceLocation = new ServiceTracker<>(context, filter, null);
-		instanceLocation.open();
-
-		try {
-			filter = context.createFilter(Location.USER_FILTER);
-		} catch (InvalidSyntaxException e) {
-			// ignore this.  It should never happen as we have tested the above format.
-		}
-		userLocation = new ServiceTracker<>(context, filter, null);
-		userLocation.open();
-
-		try {
-			filter = context.createFilter(Location.CONFIGURATION_FILTER);
+			filter = context.createFilter(filterStr);
```

Comment:
Maybe use FrameworkUtil.createFilter instead

Reply:
I can't see a real improvement over context.createFilter() and the drawback would be that the containers configuration DEBUG_FILTER setting is not considered anymore.
I don't think that is a very important feature since those filters are actually testsed during development, but because FrameworkUtil.create() does not have a clear advantage, I would leave it as it is.

Reply:
FrameworkUtil.createFilter do not need a bundlecontext and can be called anywhere, even creating a constant. If its assumed you never change the filter, what DEBUG_FILTER is good for?

Reply:
> FrameworkUtil.createFilter do not need a bundlecontext and can be called anywhere, even creating a constant. If its assumed you never change the filter, what DEBUG_FILTER is good for?

Probably for nothing. :) But since the context is needed in this method anyway we don't gain much in using FrameworkUtil.
However, I changed it to use `FrameworkUtil`.